### ISSUE TITLE
County/chapman_code inconsistency

### ISSUE THREAD

**Issue by Captainkirkdawson** [FreeUKGen/MyopicVicar#789]
The freereg1_csv_files and freereg1_csv_entries collection use county as chapman_code this leads to potential problems as all other collections use chapman_code for chapman_code and county is the name version of the code

**Comment by PatReynolds** [comment]
Is this something for Mike Fry to address?

**Comment by Captainkirkdawson** [comment]
No; this is an internal coding issue

**Comment by PatReynolds** [comment]
@Captainkirkdawson is this still an issue, and if so, is it to be resolved as part of a wider consideration of geographic locations, or is it separate?

**Comment by Sherlock21** [comment]
I have no problem changing to use WORDS instead of the 3 digit Chapman Code except for the space the County name will take up on screen and in search results.
Also: remember that in Marriages (in FR2) if the record is for County = say: DEV, which becomes Devon in this proposal, and the Groom of the Marriage is from Taunton - say- then his parish is recorded as: "Taunton (SOM)"
So we need to think out whether to leave these cases as they presently are, or run a fix to change all such to the text version 'County Name' instead of the Chapman code version.

**Comment by edickens** [comment]
If I understand this correctly, this is how the system works internally.  A problem would arise if the County Name was changed.  We did have a change but I can't remember which.
If the Name is capable of being changed, then a lot of other things will follow, so perhaps it should not be possible to change the name.  Ideally I think the Chapman code should be used at all times.

**Comment by Captainkirkdawson** [comment]
Yes it remains an issue. It is an internal issue and has no bearing on EricB's comment.  
It is an issue that will likely bite a developer who is unaware of the inconsistency. 
It was an error dating back to the days of FR1 and carried forward into FR2.  
It arises because the county field in a csv file is actually a chapman_code. 
We should have named the field in FR2 as chapman_code but did not; not thinking ahead we used the long standing field name county. Subsequently we started using a field named county everywhere else to be the word name for a county. The entries collection is the only one to place chapman_code in the field named county.
The likely best solution is in the entries model we should include the following
alias_attribute :chapman_code, :county
Then should a future developer use entry.chapman_code they will access entry.county

**Comment by stsccfr** [comment]
This issue has been addressed in #2760
